src/test/java/org/java_websocket/issues/Issue900Test.java

```diff
@@ -0,0 +1,155 @@
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+package org.java_websocket.issues;
+
+import org.java_websocket.WebSocket;
+import org.java_websocket.WebSocketImpl;
+import org.java_websocket.WrappedByteChannel;
+import org.java_websocket.client.WebSocketClient;
+import org.java_websocket.handshake.ClientHandshake;
+import org.java_websocket.handshake.ServerHandshake;
+import org.java_websocket.server.WebSocketServer;
+import org.java_websocket.util.SocketUtil;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+
+public class Issue900Test {
+
+    CountDownLatch countServerDownLatch = new CountDownLatch(1);
+    CountDownLatch countCloseCallsDownLatch = new CountDownLatch(1);
+
+    @Test(timeout = 2000)
+    public void testIssue() throws Exception {
+        int port = SocketUtil.getAvailablePort();
+        final WebSocketClient client = new WebSocketClient(new URI("ws://localhost:" + port)) {
+            @Override
+            public void onOpen(ServerHandshake handshakedata) {
+
+            }
+
+            @Override
+            public void onMessage(String message) {
+            }
+
+            @Override
+            public void onClose(int code, String reason, boolean remote) {
+            }
+
+            @Override
+            public void onError(Exception ex) {
+
+            }
+        };
+        WebSocketServer server = new WebSocketServer(new InetSocketAddress(port)) {
+            @Override
+            public void onOpen(WebSocket conn, ClientHandshake handshake) {
+            }
+
+            @Override
+            public void onClose(WebSocket conn, int code, String reason, boolean remote) {
+                countCloseCallsDownLatch.countDown();
+            }
+
+            @Override
+            public void onMessage(WebSocket conn, String message) {
+
+            }
+
+            @Override
+            public void onError(WebSocket conn, Exception ex) {
+
+            }
+
+            @Override
+            public void onStart() {
+                countServerDownLatch.countDown();
+            }
+        };
+        new Thread(server).start();
+        countServerDownLatch.await();
+        client.connectBlocking();
+        WebSocketImpl websocketImpl = (WebSocketImpl)new ArrayList<WebSocket>(server.getConnections()).get(0);
+        websocketImpl.setChannel(new ExceptionThrowingByteChannel());
+        server.broadcast("test");
+        countCloseCallsDownLatch.await();
+    }
+    class ExceptionThrowingByteChannel implements WrappedByteChannel {
+
+        @Override
+        public boolean isNeedWrite() {
+            return false;
```

Comment:
Shouldn't this return true? Looking at https://github.com/TooTallNate/Java-WebSocket/blob/ec0fa03176a316b1f79f2bb208493659f4c70e6b/src/main/java/org/java_websocket/SocketChannelIOHelper.java#L88-L90

it seems that the `writeMore()` isn't actually called in this test. Or are you relying on other methods to throw the exception?

Reply:
yeah, good point, I relied on the normal read/write call. Changing readMore/writeMore to true will however ensure better behavior